### PR TITLE
feat: integrate chain_id and protocol_version in NetworkInfo

### LINKED ISSUE #980: Integrate `chain_id` and `protocol_version` in `NetworkInfo`
### Describe the feature

for the `eth` namespace we need:

https://github.com/paradigmxyz/reth/blob/681f8a601a7eb4ac9141ebcbf37ba8d52d1f41e0/crates/net/rpc/src/eth/api/mod.rs#L62-L72

this info should be provided via the `NetworkInfo` trait.

I think chain_id should be a separate function and protocol version perhaps part of the `NetworkStatus`.

For chain_id we could create an `Arc<AtomicU64>` in `NetworkInner`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by leruaa**
Closes #980

### CODE REVIEW

File: crates/net/rpc/src/eth/api/mod.rs [hunk 1]
```diff
@@ -43,17 +47,23 @@ where
     Client: BlockProvider + StateProviderFactory + 'static,
 {
     /// Creates a new, shareable instance.
-    pub fn new(client: Arc<Client>, pool: Pool) -> Self {
-        let inner = EthApiInner { client, pool };
+    pub fn new(client: Arc<Client>, pool: Pool, network: NetworkHandle) -> Self {
```
Comment (mattsse) [inline crates/net/rpc/src/eth/api/mod.rs:50]: please have a look at #979 

we only want traits here and not `NetworkHandle`

File: crates/net/rpc/src/eth/api/mod.rs [hunk 2]
```diff
@@ -43,17 +47,23 @@ where
...
+        let inner = EthApiInner { client, pool, network };
         Self { inner: Arc::new(inner) }
     }
 
     /// Returns the inner `Client`
     fn client(&self) -> &Arc<Client> {
         &self.inner.client
     }
+
+    /// Returns the inner `Client`
```
Comment (mattsse) [inline crates/net/rpc/src/eth/api/mod.rs:60]: update docs

**Review by mattsse — CHANGES_REQUESTED** [review]
thanks, looks great overall, only a few nits

File: crates/net/rpc/src/eth/api/mod.rs (hunk 1 above)
Comment (leruaa) [reply crates/net/rpc/src/eth/api/mod.rs:50]: Is is ok to add `chain_id()` to `NetworkInfo` or do you prefer to create another trait?

Comment (mattsse) [reply crates/net/rpc/src/eth/api/mod.rs:50]: yep let's add this

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**